metaflow/plugins/argo/argo_workflows_cli.py

```diff
@@ -511,3 +501,81 @@ def _convert_value(param):
         "(run-id *{run_id}*).".format(name=obj.workflow_name, run_id=run_id),
         bold=True,
     )
+
+
+@argo_workflows.command(help="Delete the Argo workflow")
+@click.option(
+    "--authorize",
+    default=None,
+    type=str,
+    help="Authorize the deletion with a production token",
+)
+@click.pass_obj
+def delete(obj, authorize=None):
+    validate_token(obj.workflow_name, obj.token_prefix, obj, authorize, "delete")
+    obj.echo("Deleting workflow *{name}*...".format(name=obj.workflow_name))
+    schedule_deleted = ArgoWorkflows.remove_schedule(obj.workflow_name)
+    if schedule_deleted:
+        obj.echo("   Removed schedule")
+
+    response = ArgoWorkflows.delete(obj.workflow_name)
+    if response:
+        obj.echo(
+            "   Workflow is being deleted\n"
+            "Deletion of resources might take a while. "
+            "Deploying a flow with the same name during this time "
+            "will fail until the deletion has completed."
+        )
+
+
+def validate_token(name, token_prefix, obj, authorize, cmd_name):
```

Comment:
Is there any commonality to be had between this function and existing `resolve_token()` [here](https://github.com/Netflix/metaflow/blob/09a2821cb143bbe283f6b70c6d807afdfad48ec5/metaflow/plugins/argo/argo_workflows_cli.py#L378)?